src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java

```diff
@@ -190,7 +190,7 @@ public class JavadocTypeCheck
 
     /** Pattern to match type name within angle brackets in javadoc param tag. */
     private static final Pattern TYPE_NAME_IN_JAVADOC_TAG =
-            Pattern.compile("\\s*<([^>]+)>.*");
+            Pattern.compile("^<([^>]+)");
```

Comment:
Addition of `^` here is what fixes this issue.

**Finding nemo:**
Let's consider the input submitted by author of the issue:
```java
/**
 * One Transaction.
 *
 * @param transactionId  unique ID of the transaction in format: {@code <first Code>:<second Code>}
 */
 ```
 
The pattern in question is used by the following method to match for generic tag type name:
https://github.com/checkstyle/checkstyle/blob/fa8b95c964251f2156db988c8d05215ea0a80368/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java#L496

The pattern is used on the tag's `firstArg`, which in this case would be:
`transactionId  unique ID of the transaction in format: {@code <first Code>:<second Code>}`

The old pattern matched angular tag type content anywhere in the line, matching `<first Code>` here, wrongly identifying it it as generic type parameter (which is not) and returning `first Code` as the param name, but the param here is `transactionId`.

The param name is always at the beggining of the tag's `firstArg`, therefore matching the pattern at the begging of line solved the problem.